firmware/Sources/MID/MID_DABS/mid_dabs.c

```diff
@@ -0,0 +1,358 @@
+/*********************************************************************************
+* @file           : mid_dabs.c
+* @brief          : Implementation of MID_DABS
+***********************************************************************************/
+
+/**********************************************************************************/
+/*                  Include common and project definition header                  */
+/**********************************************************************************/
+#include "hal_adc.h"
+#include "hal_sts.h"
+#include "hal_gpio.h"
+#include "epc_conf.h"
+
+/**********************************************************************************/
+/*                        Include headers of the component                        */
+/**********************************************************************************/
+#include "mid_dabs.h"
+/**********************************************************************************/
+/*                              Include other headers                             */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                     Definition of local symbolic constants                     */
+/**********************************************************************************/
+#define MAX_LEDS_STEPS 5
+#define MAX_HS_V 15000
```

Comment:
Those defines should be taken from EPC_CONF registers

Reply:
Change math operation to factors + offset registers imported from epc_conf